src/test/java/ch/wisv/areafiftylan/integration/TicketRestIntegrationTest.java

```diff
@@ -104,6 +109,75 @@ public void testGetAllTicketsAsAdmin() {
         //@formatter:on
     }
 
+    @Test
+    public void testdeleteTicketAsAdmin() {
+        User admin = createAdmin();
+        User ticketOwner = createUser();
+        Ticket ticket = createTicketForUser(ticketOwner);
+
+        //@formatter:off
+            given().
+                header(getXAuthTokenHeaderForUser(admin)).
+            when().
+                delete(TICKETS_ENDPOINT + "/" + ticket.getId()).
+            then().
+                statusCode(HttpStatus.SC_OK);
+        //@formatter:on
+
+        assertThat(ticketService.getAllTickets()).doesNotContain(ticket);
+    }
+
+    @Test
+    public void testdeleteTicketAsAnon() {
+        User ticketOwner = createUser();
+        Ticket ticket = createTicketForUser(ticketOwner);
+
+        //@formatter:off
+            when().
+                delete(TICKETS_ENDPOINT + "/" + ticket.getId()).
+            then().
+                statusCode(HttpStatus.SC_FORBIDDEN);
+        //@formatter:on
+    }
+
+    @Test
+    public void testdeleteTicketAsUser() {
+        User user = createUser();
+        User ticketOwner = createUser();
+        Ticket ticket = createTicketForUser(ticketOwner);
+
+        //@formatter:off
+            given().
+                header(getXAuthTokenHeaderForUser(user)).
+            when().
+                delete(TICKETS_ENDPOINT + "/" + ticket.getId()).
+            then().
+                statusCode(HttpStatus.SC_FORBIDDEN);
+        //@formatter:on
+    }
+
+    @Test
+    public void testdeleteTicketAsAdminInOrder() {
+        User ticketOwner = createUser();
+        TicketType type =
+                new TicketType("testAddType", "Type for adding test", 10, 0, LocalDateTime.now().plusDays(1), false);
+        ticketService.addTicketType(type);
+        Ticket ticket = new Ticket(ticketOwner, type);
+
+        Order order = orderService.create("testAddType", null);
+        User admin = createAdmin();
+
+        //@formatter:off
+            given().
+                header(getXAuthTokenHeaderForUser(admin)).
+            when().
+                delete(TICKETS_ENDPOINT + "/" + ticket.getId()).
```

Comment:
This is a Bad Request because this is the local ticket (that has no ID yet). Try removing the actual ticket in the order:
```
Ticket ticket = order.getTickets().stream().findFirst().orElseThrow();
```
(this will result in a different statusCode)

Reply:
Thank you, updated it.